go/arrow/csv/writer.go

```diff
@@ -127,3 +127,19 @@ func (w *Writer) Write(record array.Record) error {
 
 	return w.w.WriteAll(recs)
 }
+
+// WriteHeader writes a header as one row to the CSV file
+func (w *Writer) WriteHeader(header string) error {
```

Comment:
I wouldn't expose `WriteHeader`: a header is something that has a particular meaning, and is present only once at the top of a CSV file.
best to keep this close to ourselves and don't let users possibly mess with it :)

also, what you've implemented isn't writing a header, it's more like adding a comment.

a header, in CSV speak, is the first non-comment line that gives the name of each column.
for the example/test we use, what the header should look like is:
```
bool;i8;i16;i32;i64;u8;u16;u32;u64;f32;f64;str
```
(if `;` is the `w.w.Comma`)

the CSV header can thus be completely infered and automatically generated from the `w.schema`.

the way I had envisionned it was to have a `WithHeader(bool)` option (for both `Reader` and `Writer`) that would automatically trigger the generation of the header in the `Writer` case (but only for the first batch of records, and only once) and the deciphering of the header (optionally making sure it matches the schema) in the `Reader` case.

finally: I believe a `WriteComment(string) error` method would be useful (but the string passed to it would be automatically prepended with the correct `w.w.Comment` rune.
not sure if `WriteComment` should be able to handle possible embedded `\n` characters inside of the comment though (or multiline comments.)

Reply:
Sorry I misunderstand header vs comment. Let me open a separate JIRA for comment and address separately.

I will work on the WithHeader support